### ISSUE TITLE
Final may only be used as the outermost type in assignments or variable annotations. 

### ISSUE THREAD

**Issue by weakish** [microsoft/pyright#590]
Pyright does not report error on `Final` used in positions other than the outermost type in assignments or variable annotations.

```python
x: List[Final[int]] = []  # no error with Pyright

def fun(x: Final[List[int]]) ->  None:  # no error with Pyright
    pass
```

The code sample above is copied from [PEP 591]:

> Final may only be used as the outermost type in assignments or variable annotations.
> Using it in any other position is an error. In particular, Final can't be used in annotations for function arguments:

> ```python
> x: List[Final[int]] = []  # Error!
>
> def fun(x: Final[List[int]]) ->  None:  # Error!
>    ...
> ```

[PEP 591]: https://www.python.org/dev/peps/pep-0591/#semantics-and-examples

Pyright 1.1.29 under vscode

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of Pyright.

**Comment by erictraut** [comment]
This is implemented in 1.1.30
